flaskbb/forum/views.py

```diff
@@ -394,68 +395,48 @@ def post(self, forum_id, slug=None):  # noqa: C901
         ids = request.form.getlist("rowid")
         tmp_topics = Topic.query.filter(Topic.id.in_(ids)).all()
 
-        if not len(tmp_topics) > 0:
+        if not Permission(IsAtleastModeratorInForum(forum=forum_instance)):
+            flash(
+                _("You do not have the permissions to execute this action."),
+                "danger",
+            )
+
+        elif not len(tmp_topics) > 0:
```

Comment:
Good job remembering to move the permissions checks in here as well. I think they make more sense here than in the TopicManager class. I see you also consolidated all the `return redirect(mod_forum_url)` calls into a single one at the end - smart! 🎓